src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -262,7 +262,7 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
                 var foundAwait = result.Any(pending => pending.Branch?.Kind == BoundKind.AwaitExpression);
                 if (!foundAwait)
                 {
-                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.Locations[0]);
+                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.NarrowLocation());
```

Comment:
@CyrusNajmabadi It seems that `MakeMethodSynchronous` refactoring depends on the diagnostic and its location.

Does it make sense if I fix `AbstractMakeMethodSynchronousCodeFixProvider.FixNodeAsync` to understand the new error location?